### PR TITLE
Modify rule S1172: Match the behaviour of TypeScript compiler for JS/TS

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**
We now consider unused parameters regardless of whether they appear before used parameters. Furthermore, we ignore intentionally ignored parameters, that is, parameters whose names start with an underscore.

### CODE REVIEW

**Review by vilchik-elena — CHANGES_REQUESTED** [review]
I believe just example with `_` is not enough, we need to describe the reasoning behind it

File: rules/S1172/javascript/rule.adoc [hunk 1]
```diff
@@ -18,39 +18,36 @@ function doSomething(b) {
...
+[source,javascript]
+----
+function doSomething(a, b, c) {
+  compute(arguments);
+}
 ----
 
-Further, when ``++arguments++`` is used in the function body, no parameter is reported as unused. 
-
+Also the rule ignores all parameters whose name starts with an underscore (``++_++``).
```
Comment (vilchik-elena) [inline rules/S1172/javascript/rule.adoc:44]: @yassin-kammoun-sonarsource wdyt?

Comment (yassin-kammoun-sonarsource) [reply rules/S1172/javascript/rule.adoc:44]: I would add a comma after `Also`, and the suggestion is on the wrong line, but still, I like it.

Comment (yassin-kammoun-sonarsource) [reply rules/S1172/javascript/rule.adoc:44]: Never mind, I missed your commit.

**Review by vilchik-elena — APPROVED** [review]

**Pull request merged**